subsys/net/lib/sockets/sockets.c

```diff
@@ -724,14 +718,66 @@ Z_SYSCALL_HANDLER(zsock_fcntl, sock, cmd, flags)
 }
 #endif
 
-int zsock_poll_internal(struct zsock_pollfd *fds, int nfds, int timeout)
+static int zsock_poll_prepare_ctx(struct net_context *ctx,
```

Comment:
I find this signature somewhat adhoc on the current implementation. Wanted to propose to change, but nevermind.

Reply:
But note that the other way is to pass structure with all these args, which would avoid too much va_list'ing. But I'm not sure it will make things better (more optimal). Please don't change unless you're sure that would be better.

Reply:
I was thinking about having a strucutre to wrap the parameters. Yet I figured that in order to make it right I would need to specify it in `fdtable.h`, yet creating dependency to `socket.h` there (`zsock_pollfd`) so I dropped the idea.